src/MIDebugEngine/AD7.Impl/AD7Disassembly.cs

```diff
@@ -162,6 +173,64 @@ public int Read(uint dwInstructions, enum_DISASSEMBLY_STREAM_FIELDS dwFields, ou
                     }
                 }
 
+                if (idx == 0 && m_lastSourceInfoStale)
+                {
+                    m_lastSourceInfoStale = false;
+
+                    ulong previousInstructionAddress = instruction.Addr - 1;
+
+                    if (previousInstructionAddress >= 0 && previousInstructionAddress < (ulong)prgDisassembly.Length)
+                    {
+                        DisassemblyData previousInstruction = prgDisassembly[previousInstructionAddress];
+                        m_dwLastSourceLine = (int)previousInstruction.posBeg.dwLine - 1;
+                        _pLastDocument = previousInstruction.bstrDocumentUrl;
+                    }
+                }
+
+                if ((dwFields & enum_DISASSEMBLY_STREAM_FIELDS.DSF_POSITION) != 0)
```

Comment:
>) [](http://example.com/codeflow?start=81&length=1)

I am assuming `instruction.Line` is 0 if it isn't set? If so, you need `&& instruction.Line != 0`